### DISCUSSION TITLE
Connection to ejabberd getting lost

### DISCUSSION THREAD

**Discussion by pavan-bandodekar** [processone/ejabberd#3851] [Questions] [answered]
Hi,

I have implemented real time notifications using ejabberd server(version: 16.01-2) and stophe.js. I am facing issue of connection to the server getting lost if it is in idle state.

Please help if anyone has any idea about what can cause this issue.
Thanks !!

**Comment by prefiks** [comment, marked as answer]
Generally on websocket should occasionally send ping to client to verify that connection is alive, so they aren't really truly idle.

Possilby tunning toplevel `websocket_timeout` option will do something (i think it's set to 300 seconds by default, you could try increasing it).

**Reply by pavan-bandodekar** [reply]
@prefiks : Thanks for the answer. I want to give a try for this. Do I need to set websocket_timeout in ejabberd.yml?

**Reply by badlop** [reply]
It seems [websocket_timeout](https://docs.ejabberd.im/admin/configuration/toplevel/#websocket-timeout) is a toplevel option, you must set it in ejabberd.yml like any other toplevel option

**Reply by pavan-bandodekar** [reply]
Thanks badlop, let me try it.
